src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -39,23 +65,21 @@ internal abstract partial class AbstractObjectBrowserLibraryManager : AbstractLi
         private object _classMemberGate = new object();
 
         private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _streamingPresenters;
+        private readonly Lazy<IComponentModel> _componentModel;
 
         protected AbstractObjectBrowserLibraryManager(
-            string languageName, 
-            Guid libraryGuid, 
-            __SymbolToolLanguage preferredLanguage, 
+            string languageName,
+            Guid libraryGuid,
+            __SymbolToolLanguage preferredLanguage,
             IServiceProvider serviceProvider)
             : base(libraryGuid, serviceProvider)
         {
             _languageName = languageName;
             _preferredLanguage = preferredLanguage;
+            _serviceProvider = serviceProvider;
 
-            var componentModel = (IComponentModel)serviceProvider.GetService(typeof(SComponentModel));
-            this.Workspace = componentModel.GetService<VisualStudioWorkspace>();
-            this.LibraryService = this.Workspace.Services.GetLanguageServices(languageName).GetService<ILibraryService>();
-            this.Workspace.WorkspaceChanged += OnWorkspaceChanged;
-
-            this._streamingPresenters = componentModel.DefaultExportProvider.GetExports<IStreamingFindUsagesPresenter>();
+            _componentModel = new Lazy<IComponentModel>(() => _serviceProvider.GetService(typeof(SComponentModel)) as IComponentModel);
+            _streamingPresenters = _componentModel.Value.DefaultExportProvider.GetExports<IStreamingFindUsagesPresenter>();
```

Comment:
i don't see how this can help.  _componentModel was made lazy... but on the next line, ```.Value``` is called on it, so the lazy does nothing...